nix/nim-status/default.nix

```diff
@@ -0,0 +1,150 @@
+  # Create a single header for multiple target archs
+  # by utilizing C preprocessor conditionals
+  buildIos = buildMap: name:
+    let
+      headerIos = writeText "${name}.h" ''
+        #if TARGET_CPU_X86_64
+        ${builtins.readFile "${buildMap.x86}/${name}.h"}
+        #elif TARGET_CPU_ARM
+        ${builtins.readFile "${buildMap.arm}/${name}.h"}
+        #else
+        ${builtins.readFile "${buildMap.arm64}/${name}.h"}
+        #endif
+      '';
+    in stdenv.mkDerivation {
+      inherit xcodeWrapper;
+      buildInputs = [ pkgs.coreutils ];
+      name = "${name}-ios-builder";
+      builder = writeScript "${name}-ios-builder.sh"
+      ''
+        export PATH=${pkgs.coreutils}/bin:${xcodeWrapper}/bin:$PATH
+        mkdir $out
+
+        # lipo merges arch-specific binaries into one fat iOS binary
+        lipo -create ${buildMap.x86}/lib${name}.a \
+             ${buildMap.arm}/lib${name}.a \
+             ${buildMap.arm64}/lib${name}.a \
+             -output $out/lib${name}.a
+
+        cp ${headerIos} $out/${name}.h
+        ${if name=="nim_status" then "cp ${buildMap.arm64}/nimbase.h $out" else ""}
+      '';
+  };
+in rec {
+  nim-status = {
+    android = {
+      x86 = buildNimStatus "android" "386";
+      arm = buildNimStatus "androideabi" "arm";
+      arm64 = buildNimStatus "android" "arm64";
+    };
+    ios = {
+      x86 = buildNimStatus "ios" "386";
+      arm = buildNimStatus "ios" "arm";
+      arm64 = buildNimStatus "ios" "arm64";
+    };
+  };
+
+  status-go = {
+    android = {
+      x86 = buildStatusGo "android" "386";
+      arm = buildStatusGo "androideabi" "arm";
+      arm64 = buildStatusGo "android" "arm64";
+    };
+    ios = {
+      x86 = buildStatusGo "ios" "386";
+      arm = buildStatusGo "ios" "arm";
+      arm64 = buildStatusGo "ios" "arm64";
+    };
+  };
+
+
+  nim-status-android = buildAndroid nim-status.android "nim_status";
+  nim-status-ios = buildIos nim-status.ios "nim_status";
+
+  status-go-android = buildAndroid status-go.android "status";
+  status-go-ios = buildIos status-go.ios "status";
+
+  android = stdenv.mkDerivation {
+      buildInputs = [ pkgs.coreutils ];
+      name = "nim-status-go-android-builder";
+      builder = writeScript "nim-status-go-android-builder.sh"
+      ''
+        export PATH=${pkgs.coreutils}/bin:$PATH
+        mkdir $out
+        for arch in "x86" "armeabi-v7a" "arm64-v8a"; do
+          mkdir $out/$arch
+
+          for filename in ${nim-status-android}/$arch/*; do
+            ln -sf "$filename" $out/$arch/$(basename $filename)
+          done
+
+          for filename in ${status-go-android}/$arch/*; do
+            ln -sf "$filename" $out/$arch/$(basename $filename)
+          done
+        done
+      '';
+  };
+
+  ios = stdenv.mkDerivation {
+      buildInputs = [ pkgs.coreutils ];
+      name = "nim-status-go-ios-builder";
+      builder = writeScript "nim-status-go-ios-builder.sh"
+      ''
+        export PATH=${pkgs.coreutils}/bin:$PATH
```

Comment:
Actually I think a cleaner way of doing this is indeed adding `coreutils` into `buildInputs`, and then using this:
```
source $stdenv/setup
```
At the beginning of the builder script.